starfish/image/_stack.py

```diff
@@ -731,8 +906,40 @@ def squeeze(self) -> numpy.ndarray:
 
     def un_squeeze(self, stack):
         if type(stack) is list:
-            stack = numpy.array(stack)
+            stack = np.array(stack)
 
         new_shape = (self.num_hybs, self.num_chs, self.num_zlayers) + self.tile_shape
         res = stack.reshape(new_shape)
         return res
+
+    @classmethod
+    def from_numpy_array(cls, array: np.ndarray) -> "ImageStack":
+        """Create an ImageStack from a 5d numpy array with shape (n_hyb, n_ch, n_z, x, y)
+
+        Parameters
+        ----------
+        array : np.ndarray
+            5-d tensor of shape (n_hyb, n_ch, n_z, x, y)
+
+        Returns
+        -------
+        ImageStack :
+            array data stored as an ImageStack
+
+        """
+        if not len(array.shape) == 5:
+            raise ValueError('a 5-d tensor with shape (n_hyb, n_ch, n_z, x, y) must be provided.')
```

Comment:
nit: inconsistency between x, y and height, width.

also, here we are doing x before y, whereas when you extract on L932, it's height and then width.

Reply:
Oh yikes, yes these instructions are wrong. We should be doing z, y, x everywhere.